packages/monorepo-utils/src/project-utils/filter.ts

```diff
@@ -1,14 +1,21 @@
 import type { Project } from '../project/project';
+import { getExportsSourceDirs } from './getExportsSourceDirs';
 
 export type Filter = FilterFunction;
 export type FilterFunction = (
   projects: Project[],
 ) => Project[] | Promise<Project[]>;
 
-export const defaultFilter: FilterFunction = projects => projects;
+export const defaultFilter: FilterFunction = (projects) => projects;
 
 export const filterByField =
-  (fieldName: string): FilterFunction =>
+  (fieldName: string, checkExports?: boolean): FilterFunction =>
   (projects: Project[]) => {
-    return projects.filter(p => fieldName in p.metaData);
+    return projects.filter((p) => {
+      return (
+        fieldName in p.metaData ||
+        (checkExports &&
+          getExportsSourceDirs(p.metaData.exports || {}, fieldName).length)
```

Comment:
Call `getExportsSourceDirs` has some performance overhead here. For example, we do not need to call `path.normalize` in this case.